src/main/java/com/gluonhq/NativeBaseMojo.java

```diff
@@ -74,14 +77,31 @@ public abstract class NativeBaseMojo extends AbstractMojo {
 
     private static final List<String> ALLOWED_DEPENDENCY_TYPES = Collections.singletonList("jar");
 
-    // TODO: Remove this restriction when MavenArtifactResolver works with Maven 3.9.0+
-    private static final Version MAX_SUPPORTED_MAVEN_VERSION = new Version(3, 8, 8);
+    private static final Version MAX_SUPPORTED_MAVEN_VERSION = new Version(3, 9, 9);
```

Comment:
Do we expect this not to work on 4?

Reply:
I haven't tested it yet, honestly. Many parts are moving and being deprecated/removed for Maven 4, so I'd say we'll need to revisited when people start to use it (beta for now), or it gets finally released.

Reply:
I believe we need to change at least some logic here, because at this point, the plugin won't work starting with maven 3.9.10. We can perhaps invert the logic and say that we support all versions up to 4.0.0 (but not inclusive)?

Reply:
Maven 4.0.0 is not released yet, but I'm trying to test Maven 4.0.0-alpha-13, and `new Version()` fails to parse it.

Removing the whole version check works (I've tested 3.6.3, 3.8.1, 3.8.8, 3.9.0, 3.9.9, 4.0.0-alpha-13), so we don't need it anymore.